metamask-predicate/src/main.sw

```diff
@@ -0,0 +1,13 @@
+predicate;
+
+use std::{b512::B512, constants::ZERO_B256, ecr::ec_recover_address, inputs::input_predicate_data};
+
+fn main(msg_hash: b256, signature: B512, expected_public_key: b256) -> bool {
```

Comment:
Shouldn't `expected_public_key` be a compile time constant? If it's just a provided parameter then I could authorize this predicate with any identity.